### DISCUSSION TITLE
User Settings are ignored - why?

### DISCUSSION THREAD

**Discussion by TomFuch** [rspamd/rspamd#3885] [Q&A]
Hello!

I work with some usersettings as described [here](https://rspamd.com/doc/configuration/settings.html). The following setting is ignored by Rspamd. The Symbol always get the default Value of 3. 
Does anybody know why?

`
userSetting1623393729618 {
    id = "reject_undisclosed_recipients";
    priority = high;
    from = "@gmail.com";
    apply {
        R_UNDISC_RCPT = 10.0;
    }
}
`

**Comment by fatalbanana** [comment]
It works for me as described; maybe logs have some hints (some other settings applied, or)... You can set `debug_modules = ["settings"]` in `local.d/logging.inc` for extra verbosity.

Your particular use-case might be better served by something like [composites](https://rspamd.com/doc/configuration/composites.html) (and [rules](https://rspamd.com/doc/tutorials/writing_rules.html) or [multimap](https://rspamd.com/doc/modules/multimap.html)).
